### ISSUE TITLE
GeoJSON dissolve output in the browser differs from node module

### ISSUE THREAD

**Issue by martinlowinski** [mbloch/mapshaper#220]
Hello,

I am currently using mapshaper to dissolve postcode areas into larger regions for a leaflet map. The postcode areas were extracted from OpenStreetMap and published on [Github](https://github.com/yetzt/postleitzahlen).

In a first step I created the geojson regions data using the node module and `mapshaper postcodes.geojson -join regions.csv keys=postcode,postcode -dissolve region -o regions.geojson`. This worked really fine, no issues here.

The next step was to generate the regions data in the browser, somewhere in the backend. So I used browserify to create the bundle using `browserify -r ./mapshaper.js:mapshaper > mapshaper-lib.js` and executed the same command. However, the resulting geojson data differs from the output of the node module. There are lots of very small lines/areas where the dissolve didn't work correctly (or at least different).

![screenshot from 2017-09-21 13-11-05](https://user-images.githubusercontent.com/1047373/30693266-0dd72a26-9ecf-11e7-81c0-c71f709dfcbf.png)

The code to generate the regions is the following:
```
var mapshaper = require('mapshaper');

// Both files are loaded via a jQuery ajax call
const input = {
  'postcodes.geojson': json,
  'regions.csv': csv,
};

mapshaper.applyCommands(
  '-i postcodes.geojson -join regions.csv keys=postcode,postcode -dissolve region -o format=geojson',
  input,
  function(err, output) {
    var data = jQuery.parseJSON(output['postcodes.json']);
  }
);
```
Any ideas what the issue is?

Thanks!

**Comment by mbloch** [comment]
Hi, I can't explain the difference -- if you used the same version of mapshaper both times, the same commands and the same input files, I would expect the same output. You should send me your input files if you'd like me to try to reproduce what you saw.

I found a single data file in the repo that you linked to: `postleitzahlen.geojson` -- is that the file that you tested with both times? That file is not 'clean' -- there are many intersections and gaps between adjacent polygons. You can verify that by dropping it into the web UI -- you'll see thousands of red dots, marking intersections, like this:
![image](https://user-images.githubusercontent.com/1232222/30698039-c9231734-9eae-11e7-999b-e27aaa803982.png)

The tiny sliver polygons that you reported are what I would expect to see after dissolving a file like this.

The best you can do in this case is to filter out the slivers, like this:
```
 mapshaper postcodes.geojson -join regions.csv keys=postcode,postcode -dissolve2 region -filter-slivers -o regions.geojson
```

Note the use of `-dissolve2`, which is more reliable for messy datasets, and the `-filter-slivers` command.

**Comment by martinlowinski** [comment]
Hi, thanks for the quick reply, `-dissolve2` and `-filter-slivers` works perfectly! I did not test the file `postleitzahlen.geojson` on the web UI but now I understand where the sliver polygons come from.

Actually, I only used a subset of the file because it's to big to process it in the web backend: `mapshaper postleitzahlen.geojson -filter 'parseInt(postcode) >= 70000 && parseInt(postcode) <= 77000' -o postleitzahlen-subset.geojson` -- if you try to reproduce the issue.

The regions.csv is a simple two-column file with the postcode in the first column and an area name in the second column to group them:
```
70437,Nord-Stuttgart
70439,Nord-Stuttgart
70469,Nord-Stuttgart
70372,Stuttgart-Bad Cannstatt
70374,Stuttgart-Bad Cannstatt
```

Thanks again!
